staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -139,6 +142,19 @@ func newStore(c *clientv3.Client, codec runtime.Codec, newFunc, newListFunc func
 	w.getCurrentStorageRV = func(ctx context.Context) (uint64, error) {
 		return storage.GetCurrentResourceVersionFromStorage(ctx, s, newListFunc, resourcePrefix, w.objectType)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		ctx := context.Background()
+		err := etcdfeature.DefaultFeatureSupportChecker.CheckClient(ctx, c, storage.RequestWatchProgress)
+		if err != nil {
+			klog.ErrorS(err, "Couldn't check features supported by etcd. Needed to use ConsistentListFromCache")
+		}
+		// Error already checked in CheckClient
+		supports, _ := etcdfeature.DefaultFeatureSupportChecker.Supports(storage.RequestWatchProgress)
+		if !supports {
+			klog.InfoS("RequestWatchProgress feature is not supported by provided etcd or etcd is not up yet. This will prevent ConsistentListFromCache from working")
+		}
```

Comment:
With CheckClient being asynchronous it doesn't make sense to check  `Supports`, we should move the log to the CheckClient.

Reply:
Done.